tensorpack/decomposition.py

```diff
@@ -34,7 +34,9 @@ def __init__(self, data, matrix=[0], max_rr=5, method=perform_CP):
                 self.hasMatrix = True
         pass
 
-    def perform_tfac(self):
+    def perform_tfac(self, callback=None, rr_callback=1):
+        if callback:
```

Comment:
if there is no callback, you won't perform the method?

Reply:
It will just run an additional CP factorization for the rank of interest (rr_callback) before anything else. I can put it after everything if that makes more sense or just remove it entirely. I just figure since we're forced to use the Decomp object anyway, might as well provide support for r2x over iteration/runtime for everything that can use it

Reply:
Why does a callback require a different rank?

Reply:
It's to look at a single number of components to consider in factorization, since the function currently finds the R2X at every number of components up to `rrs` (the tracker callback class can only track one number of components considered in factorization at a time). I can remove it entirely to set it to the maximum number of components as well.

Reply:
I'm also not following the purpose of this. A callback is to peer within the factorization method during a run. This seems like something Decomposition could use, but the interface here doesn't make sense to me.